contracts/nftauction.js

```diff
@@ -599,3 +688,82 @@ actions.updateAuctions = async () => {
     await settleAuction(auction);
   }
 };
+
+actions.setMarketParams = async (payload) => {
+  const {
+    symbol, officialMarket, agentCut, minFee, isSignedWithActiveKey,
+  } = payload;
+
+  if (!api.assert(symbol && typeof symbol === 'string', 'invalid params')) {
+    return false;
+  }
+
+  // if no parameters supplied, we have nothing to do
+  if (officialMarket === undefined && agentCut === undefined && minFee === undefined) {
```

Comment:
There appears to be a lot of extra logic here. 

Recommendation is just to create the params up front if it does not exist,
Then go through the settings and update.

Also, I would assert errors so it's obvious if a provided parameter is malformed.

It looks like you are distinguishing between null and undefined as well, so people can unset a parameter right?